models/fhir/extension.ts

```diff
@@ -48,3 +48,10 @@ export interface ExtensionExtension<T extends Extension> extends Extension {
 export interface BooleanExtension extends Extension {
   valueBoolean: boolean
 }
+
+export type RepeatInformationExtension = ExtensionExtension<UnsignedIntExtension | DateTimeExtension>
```

Comment:
this is actually way more readable in code, maybe all our extensions should be like this

Reply:
for example, on the MedicationRequestOutcome we have: 
```extension: Array<extension.ReferenceExtension<practitionerRole.PractitionerRole>
    | extension.PrescriptionStatusHistoryExtension>
```
its easy to tell from this that it has the status history extension on it but not a clue what the other extension is